src/context.ml

```diff
@@ -426,6 +428,23 @@ let install_prefix t =
   | Some x -> Path.absolute x
   | None   -> Path.parent t.ocaml_bin
 
+let install_ocaml_libdir t =
+  (* If ocamlfind is present, it has precedence over everything else. *)
+  match which t "ocamlfind" with
+  | Some fn ->
+    (Future.run_capture_line ~env:t.env Strict
+       (Path.to_string fn) ["printconf"; "destdir"]
+     >>| Path.absolute)
+  | None ->
+    (* If there no ocamlfind in the PATH, check if we have opam and assume a standard
+       opam setup *)
+    opam_config_var t "lib"
+    >>| function
+    | Some s -> Path.absolute s
+    | None ->
+      (* If neither opam neither ocamlfind are present use stdlib dir *)
+      t.stdlib_dir
```

Comment:
I think it's a bit surprising that for the search path, jbuilder falls back to `$(which ocamlc)/../../lib` and for installation it falls back to `$(ocamlc -where)`. I don't claim that I made the right choice for the search path, happy to change it if the current default is wrong, but I think the two should be consistent